TinyJSON/Encoder.cs

```diff
@@ -146,12 +147,25 @@ void EncodeValue(object value, bool forceTypeHint, int combineIndex)
 					EncodeValue(asVariant.value, forceTypeHint, combineIndex);
 				}
 				else
+				if (IsHashSet(value))
+				{
+					EncodeHashSet(value, forceTypeHint);
+				}
+				else
 				{
 					EncodeOther(value, forceTypeHint, combineIndex);
 				}
 			}
 		}
 
+		bool IsHashSet(object value)
+		{
+			Type hashSetType = typeof(HashSet<>);
+			Type valueType = value.GetType();
+			//do not use the full name as the instantiated hashset will be longer
+			int comparison = string.Compare(hashSetType.Name, valueType.Name);
```

Comment:
This string comparison done because HashSet is a generic?

Reply:
Yes.  There's no nongeneric interface for HashSets like IDictionary or IList.

Reply:
Technically, `HashSet<T>` implement `IEnumerable` (regular and generic), but it's _too_ generic to determine if something's a `HashSet<T>`.